### PR TITLE
Fire plotly_afterplot after Promise.all callback

### PR DESCRIPTION
**Pull request by etpinard**
fixes https://github.com/plotly/plotly.js/issues/1341

### CODE REVIEW

File: test/jasmine/tests/plot_api_test.js [hunk 1]
```diff
@@ -84,6 +84,21 @@ describe('Test plot api', function() {
                 expect(gd._transitionData._frames[2].name).toEqual('frame3');
             }).catch(fail).then(done);
         });
+
+        it('should emit afterplot event after plotting is done', function(done) {
```
Comment (etpinard) [inline test/jasmine/tests/plot_api_test.js]: This test here fails on `master`.

I'm not exactly sure why though which might me related to https://github.com/plotly/plotly.js/issues/76

Consider:

```js
var gd = document.getElementById('graph')

Plotly.plot(gd, [{}])

gd.on('plotly_afterplot', () => {})
```

I suspect that on `master` the `Plotly.plot` promise is resolved before `gd.on('plotly_relayout, /* */)` is set. By tucking `gd.emit('plotly_afterplot')` is the `Promise.all` callback the `Plotly.plot` promise is resolved after `gd.on`.

Comment (alexcjohnson) [inline test/jasmine/tests/plot_api_test.js]: > I suspect that on master the Plotly.plot promise is resolved before gd.on('plotly_afterplot, /* */) is set.

exactly - in most cases the plotting itself is sync, so your fix here is spot on.

**Comment by alexcjohnson** [comment]
💃

**Comment by alexander-daniel** [comment]
Just wondering, is this only fired on `Plotly.plot` calls?

Curious, as I'd love to use this event with the React component, but that seems to use either:
`Plotly.newPlot`  &
`Plotly.react`

Thanks!

**Comment by etpinard** [comment]
Right, `plotly_afterplot` should be fired when `Plotly.newPlot` completes. Not sure why that doesn't happen.

**Pull request merged**